### ISSUE TITLE
CallbackRegistryMixin: HasRegistrantsForEvent incorrectly returns true if no registrants exist

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#177]
The [CallbackRegistryMixin:HasRegistrantsForEvent](https://github.com/Gethe/wow-ui-source/blob/49fd6c5e5e5b149ce6e63700d07d62868d8c07ef/Interface/SharedXML/CallbackRegistry.lua#L55-L57) method may incorrectly return `true` in cases where no callbacks are registered for an event.

The sequence of operations required to trigger this involve just registering and then unregistering a callback, as shown below:

```
/run EventRegistry:RegisterCallback("TestEvent", print, "UniqueOwner")
/run EventRegistry:UnregisterCallback("TestEvent", "UniqueOwner")
/run assert(not EventRegistry:HasRegistrantsForEvent("TestEvent"), "'TestEvent' supposedly still has registrants")
```

The precise issue this causes is that the callback tables for `TestEvent` above exist, but have no contents. As HasRegistrantsForEvent only ever tests to see if the table exists, it incorrectly thinks registrants exist despite no callbacks being present inside them.

A simple fix would be for this to instead check the following:

```lua
function CallbackRegistryMixin:HasRegistrantsForEvent(event)
	for callbackType, callbackTable in pairs(self:GetCallbackTables()) do
		local eventRegistrants = callbackTable[event];
		if eventRegistrants and securecallfunction(next, eventRegistrants) then
			return true;
		end
	end
	return false;
end
```

Which will cause the function to return `true` if the table exists and contains any value. A trip through `securecallfunction` would prevent this check from tainting the caller - which would allow for reference tracking of frame events on the global EventRegistry to be implemented in a way that doesn't taint (#178).

**Comment by Meorawr** [comment]
Fixed internally, pending a future build.

**Comment by Meorawr** [comment]
Fixed in 9.1.5.40383, not yet fixed in other clients.

**Comment by Meorawr** [comment]
Fixed in 1.14.1.40487, should show up in BCC eventually.

**Comment by Meorawr** [comment]
Fixed in 2.5.3.41402.
